src/ai/ai_help_structs.cc

```diff
@@ -1188,6 +1234,22 @@ uint32_t PlayersStrengths::get_enemies_max_land() {
 	return land;
 }
 
+uint32_t PlayersStrengths::get_max_power() {
+	uint32_t power = 0;
+	for (auto& item : all_stats) {
+		power = std::max<uint32_t>(power, item.second.players_power);
+	}
+	return power;
+}
+
+uint32_t PlayersStrengths::get_max_land() {
```

Comment:
Shouldn't buildings get an equivalent too?

Reply:
maybe, do you think number of buildings is that important to determine whether to team up?

Reply:
I think the number of buildings is a good indicator of the strength of a player's economy, so it's at least as important as land owned. Land owned in itself is only potential and cushion against attacks. So military, land and buildings are the 3 main indicators of a player's strength for me, all of them with about the same weight.

Reply:
ok I will implement something like this then